app/src/main/java/org/wikipedia/search/SearchResultsFragment.kt

```diff
@@ -0,0 +1,532 @@
+        override fun getItemViewType(position: Int): Int {
+            return if (totalResults.isEmpty()) VIEW_TYPE_NO_RESULTS else VIEW_TYPE_ITEM
+        }
+
+        override fun getItemCount(): Int {
+            return if (totalResults.isEmpty()) resultsCountList.size else totalResults.size
+        }
+
+        override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): DefaultViewHolder<View> {
+            return if (viewType == VIEW_TYPE_ITEM) {
+                SearchResultItemViewHolder(LayoutInflater.from(context)
+                        .inflate(R.layout.item_search_result, parent, false))
+            } else {
+                NoSearchResultItemViewHolder(LayoutInflater.from(context)
+                        .inflate(R.layout.item_search_no_results, parent, false))
+            }
+        }
+
+        override fun onBindViewHolder(holder: DefaultViewHolder<View>, pos: Int) {
+            if (holder is SearchResultItemViewHolder) {
+                holder.bindItem(pos)
+            } else if (holder is NoSearchResultItemViewHolder) {
+                holder.bindItem(pos)
+            }
+        }
+    }
+
+    private inner class NoSearchResultItemViewHolder(itemView: View) : DefaultViewHolder<View>(itemView) {
+        private val accentColorStateList = ColorStateList.valueOf(getThemedColor(requireContext(), R.attr.colorAccent))
+        private val secondaryColorStateList = ColorStateList.valueOf(getThemedColor(requireContext(), R.attr.material_theme_secondary_color))
+        fun bindItem(position: Int) {
+            val langCode = WikipediaApp.getInstance().language().appLanguageCodes[position]
+            val resultsCount = resultsCountList[position]
+            val resultsText = view.findViewById<TextView>(R.id.results_text)
+            val languageCodeText = view.findViewById<TextView>(R.id.language_code)
+            resultsText.text = if (resultsCount == 0) getString(R.string.search_results_count_zero) else resources.getQuantityString(R.plurals.search_results_count, resultsCount, resultsCount)
+            resultsText.setTextColor(if (resultsCount == 0) secondaryColorStateList else accentColorStateList)
+            languageCodeText.visibility = if (resultsCountList.size == 1) View.GONE else View.VISIBLE
+            languageCodeText.text = langCode
+            languageCodeText.setTextColor(if (resultsCount == 0) secondaryColorStateList else accentColorStateList)
+            languageCodeText.backgroundTintList = if (resultsCount == 0) secondaryColorStateList else accentColorStateList
+            formatLangButton(languageCodeText, langCode,
+                    SearchFragment.LANG_BUTTON_TEXT_SIZE_SMALLER, SearchFragment.LANG_BUTTON_TEXT_SIZE_LARGER)
+            view.isEnabled = resultsCount > 0
+            view.setOnClickListener {
+                (parentFragment as SearchFragment).setUpLanguageScroll(position)
```

Comment:
Should we use `requireParentFragment()` on this one and also add "`isAdded` to prevent fragment not being attached?